homeassistant/components/insteon/utils.py

```diff
@@ -200,29 +249,58 @@ def scene_off(service):
     hass.services.async_register(
         DOMAIN, SRV_SCENE_OFF, scene_off, schema=TRIGGER_SCENE_SCHEMA
     )
+    async_dispatcher_connect(hass, SIGNAL_SAVE_DEVICES, _save_devices)
     _LOGGER.debug("Insteon Services registered")
 
 
 def print_aldb_to_log(aldb):
     """Print the All-Link Database to the log file."""
-    _LOGGER.info("ALDB load status is %s", aldb.status.name)
+    # This service is useless if the log level is not INFO for the
+    # insteon component. Setting the log level to INFO and resetting it
+    # back when we are done
+    orig_log_level = _LOGGER.level
+    if orig_log_level > logging.INFO:
+        _LOGGER.setLevel(logging.INFO)
+    _LOGGER.info("%s ALDB load status is %s", aldb.address, aldb.status.name)
     if aldb.status not in [ALDBStatus.LOADED, ALDBStatus.PARTIAL]:
-        _LOGGER.warning("Device All-Link database not loaded")
-        _LOGGER.warning("Use service insteon.load_aldb first")
-        return
+        _LOGGER.warning("All-Link database not loaded")
 
     _LOGGER.info("RecID In Use Mode HWM Group Address  Data 1 Data 2 Data 3")
     _LOGGER.info("----- ------ ---- --- ----- -------- ------ ------ ------")
     for mem_addr in aldb:
         rec = aldb[mem_addr]
         # For now we write this to the log
         # Roadmap is to create a configuration panel
-        in_use = "Y" if rec.control_flags.is_in_use else "N"
-        mode = "C" if rec.control_flags.is_controller else "R"
-        hwm = "Y" if rec.control_flags.is_high_water_mark else "N"
+        in_use = "Y" if rec.is_in_use else "N"
+        mode = "C" if rec.is_controller else "R"
+        hwm = "Y" if rec.is_high_water_mark else "N"
         log_msg = (
             f" {rec.mem_addr:04x}    {in_use:s}     {mode:s}   {hwm:s}    "
-            f"{rec.group:3d} {rec.address.human:s}   {rec.data1:3d}   "
+            f"{rec.group:3d} {str(rec.target):s}   {rec.data1:3d}   "
             f"{rec.data2:3d}   {rec.data3:3d}"
         )
         _LOGGER.info(log_msg)
+    _LOGGER.setLevel(orig_log_level)
+
+
+@callback
+def async_add_insteon_entities(
+    hass, platform, entity_type, async_add_entities, discovery_info
+):
+    """Add Insteon devices to a platform."""
+    new_entities = []
+    device_list = [discovery_info.get("address")] if discovery_info else devices
+
+    for address in device_list:
+        device = devices[address]
+        groups = get_platform_groups(device, platform)
+        for group in groups:
+            _LOGGER.debug(
```

Comment:
Please just log the list of entities outside the loop. The entity has a repr magic method.